### ISSUE TITLE
Bug: [no-non-null-assertion] +error.svelte doesn't allow `$page.error!` to be asserted non-null

### ISSUE THREAD

**Issue by justingolden21** [typescript-eslint/typescript-eslint#6786]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

Unable to add svelte to playground

### Repro Code

```typescript
`+error.svelte`

<script lang="ts">
	import { page } from '$app/stores';

	$: error = $page.error!;
	$: status = $page.status;
</script>
```

### ESLint Config

```javascript
module.exports = {
module.exports = {
	root: true,
	parser: '@typescript-eslint/parser',
	extends: ['eslint:recommended', 'plugin:@typescript-eslint/recommended', 'prettier'],
	plugins: ['prettier', 'svelte3', '@typescript-eslint'],
	ignorePatterns: ['*.cjs'],
	overrides: [
		{
			files: ['*.svelte'],
			processor: 'svelte3/svelte3'
		}
	],
	settings: {
		'svelte3/typescript': () => require('typescript')
	}
};
```

### tsconfig

```jsonc
{
	"extends": "./.svelte-kit/tsconfig.json",
	"compilerOptions": {
		"allowJs": true,
		"checkJs": true,
		"esModuleInterop": true,
		"forceConsistentCasingInFileNames": true,
		"resolveJsonModule": true,
		"skipLibCheck": true,
		"sourceMap": true,
		"strict": true
	}
}
```

### Expected Result

No error

### Actual Result

Complains of `no-non-null-assertion` when running `npm run lint` (`"lint": "eslint .",` in `package.json`)

**Comment by bradzacher** [comment]
What does `` `+error.svelte` `` do?
I'm guessing it's a svelte-specific thing that changes how the svelte tooling works..?

**Comment by JoshuaKGoldberg** [comment]
This is a Svelte-specific thing. https://kit.svelte.dev/docs/routing#error

The Svelte ESLint plugin would be the right place for Svelte-specific logic. You'd probably have better luck filing this issue there: https://github.com/sveltejs/eslint-plugin-svelte3

If that plugin needs us to expose some new API for this use case, we'd be happy to take in an issue describing what & why.

Cheers!
